src/Symfony/Component/VarDumper/Dump.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper;
+
+use Symfony\Component\VarDumper\Cloner\VarCloner;
+use Symfony\Component\VarDumper\Dumper\CliDumper;
+use Symfony\Component\VarDumper\Dumper\HtmlDumper;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @method self to(callable|resource|string $output)               Sets the output destination of the dump: a line callable, an opened stream or an output path.
+ * @method self as(string $format)                                 Sets the output format of the dump: can be "html", "cli", "text" or "env", which uses the `DUMP_AS` env var or the PHP SAPI.
+ * @method self die()                                              Dies after dumping.
+ * @method self withTitle(string $title)                           Sets the title of the dump.
+ * @method self withTrace(bool $trace)                             Enables/disables dumping the stack trace after dumping.
+ * @method self withMaxItems(int $items)                           Limits the number of items past the first level.
+ * @method self withMaxDepth(int $depth)                           Limits the depth of the dump.
+ * @method self withMaxItemsPerDepth(int $max)                     Limits the number of items per depth level.
+ * @method self withMaxStringLength(int $length)                   Limits the number of characters for dumped strings.
+ * @method self withMaxStringWidth(int $width)                     Limits the number of characters per line for dumped strings.
+ * @method self withRefHandles(bool $handles)                      Enables/disables objects' global identifiers dumping.
+ * @method self withCasters(array $casters, bool $replace = false) Adds or replace casters for resources and objects.
+ * @method self withFilter(int $filter)                            A bit field of Caster::EXCLUDE_* constants.
```

Comment:
implssible because `as` or `die` are reserved keyworkds, and still the best name for the keywords.
Also because using `__call` removes a lot of duplications.